src/foam/dao/index/AATree.js

```diff
@@ -272,32 +260,40 @@ foam.CLASS({
       return s;
     },
 
-    function select(sink, skip, limit, order, predicate) {
+  
+    function select(sink, skip, limit, order, predicate, cache) {
       if ( limit && limit[0] <= 0 ) return;
 
       if ( skip && skip[0] >= this.size && ! predicate ) {
         skip[0] -= this.size;
         return;
       }
 
-      this.left.select(sink, skip, limit, order, predicate);
-      this.value.select(sink, skip, limit, order, predicate);
-      this.right.select(sink, skip, limit, order, predicate);
+      this.left.select(sink, skip, limit, order, predicate, cache);
+
+      this.value.select(sink, skip, limit,
+        order && order.orderTail(), predicate, cache);
+
+      this.right.select(sink, skip, limit, order, predicate, cache);
     },
 
-    function selectReverse(sink, skip, limit, order, predicate) {
+   
+    function selectReverse(sink, skip, limit, order, predicate, cache) {
       if ( limit && limit[0] <= 0 ) return;
 
-
       if ( skip && skip[0] >= this.size && ! predicate ) {
-        console.log('reverse skipping: ', this.key);
+        //console.log('reverse skipping: ', this.key);
         skip[0] -= this.size;
         return;
       }
 
-      this.right.selectReverse(sink, skip, limit, order, predicate);
-      this.value.selectReverse(sink, skip, limit, order, predicate);
-      this.left.selectReverse(sink,  skip, limit, order, predicate);
+      this.right.selectReverse(sink, skip, limit, order, predicate, cache);
+
+      // select() will pick reverse or not based on order
```

Comment:
The comparator specified will give a 1, -1 direction indicator. Select() moves to the next ordering down the chain and passes that on to subsequent tail indexes.